Tensile/Source/lib/source/ContractionSolution.cpp

```diff
@@ -1756,15 +1788,28 @@ namespace Tensile
                 dSize.z *= problem.boundSize(i);
             }
 
-            skGrid = streamk::best_predicted_grid_size(sizeMapping.macroTile.x, sizeMapping.macroTile.y, sizeMapping.depthU, dSize.x, dSize.y, dSize.z);
+            return streamk::best_predicted_grid_size(sizeMapping.macroTile.x, sizeMapping.macroTile.y, sizeMapping.depthU, dSize.x, dSize.y, dSize.z, cuCount);
+        }
+
+        // Limit the CUs Stream-K is launched on either max or the specified,
+        // whichever is minimum.
+        else if(pAMDGPU->skMaxCUs > 0) 
+        {
+            return min(cuCount, pAMDGPU->skMaxCUs);
+        }
+
+        // Multiply the cuCount with a constant factor (c), and launch
+        // c * cuCount number of workgroups for Stream-K.
+        else if(pAMDGPU->skGridMultiplier > 1) 
+        {
+            return cuCount * pAMDGPU->skGridMultiplier;
+        }
```

Comment:
Reordering this part means the environment options for Max CUs, dynamic grid and grid multiplier can no longer be used together. Maybe we don't need that behaviour anymore, cause we also have the fixed grid option for doing performance sweeps, but we have used some combos for testing in the past.

Reply:
1. Even if we set the MAX CUs and then apply dynamic grid later, we don't have CU-masking. Will it actually only apply dynamic grid selection with MAX CUs as the new CU count? Is that what you're suggesting when we mix the settings together?
2. I don't think Grid Multiplier needs to be used with other dynamic settings.